lib/download-png.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { saveAs } from "file-saver";
+
+export class DownloadPNG extends React.Component {
+    constructor(props) {
+        super(props);
+        this.saveCanvas = this.saveCanvas.bind(this);
+        this.state = {
+            data: {},
+            dataIsLoaded: false,
+            fileName: "",
+        };
+    }
+
+    componentDidMount() {
+        this.setState({
+            data: document.getElementById("canvas"),
+            dataIsLoaded: true,
+            fileName: this.props.fileName,
+        });
+    }
+
+    saveCanvas() {
+        const fileName = this.state.fileName;
+        //save to png
+        if (this.state.dataIsLoaded) {
+            const data = this.state.data;
+            data.toBlob(function (blob) {
+                saveAs(blob, `${fileName}.png`);
+            });
+        }
+    }
+
+    render() {
+        return (
+            <a
+                onClick={this.saveCanvas}
+                class='btn btn-primary float-left bg-flat-color-1'
+                title='Download as PNG'
+            >
+                <svg
+                    xmlns='http://www.w3.org/2000/svg'
```

Comment:
It'd be good to reuse the heroicons library for this

Reply:
for this one it can be the "photograph" icon. And there should also be some text, e.g. "Download Image"